### DISCUSSION TITLE
V2中如何自定义fieldState、formState？

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#1215] [Q&A 2.x] [answered]
在1.x中，可直接使用`setFieldState`、`setFormState`自定义状态。2.x如何做？
群里有回复说2.x不支持自定义状态，需要使用`@formily/reactive`，希望文档中能给出案例。

**Comment by janryWang** [comment, marked as answer]
不推荐往表单模型，字段模型里挂状态了，直接使用@formily/reactive维护状态就行了，具体案例没啥特殊的，看reactive文档即可

**Reply by NexxLuo** [reply]
为什么不推荐挂自定义状态了呢？如果挂了自定义状态会有什么影响吗？用@formily/reactive实现的话没有那么直接，也相对更麻烦一点。

**Reply by NexxLuo** [reply]
我这里有一个场景，自定义组件中有各种自定义状态，而且需要在effects中进行控制。如果不挂在字段模型中，这些自定义状态在effects中就不好获取和设置值了。

**Reply by janryWang** [reply]
那你可以挂在component props上呀
